### ISSUE TITLE
請問能繼續更新RITDarkroom這麼好的軟體嗎？

### ISSUE THREAD

**Issue by ai-gate** [FunkyRider/RITDarkroom#2]
不好意思打擾，一直在使用這一套軟體，但相機一直沒有更新…想說已經習慣這一套軟體的很好表現，不知道您可能有機會繼續更新？
付錢的話可以嗎？謝謝。

**Comment by FunkyRider** [comment]
Hi, I stopped working on this project years ago. There really wasn't any demand for adding new functionalities. Adding new manufactures can be a lot of effort but if you have an existing Canon or Panasonic camera that you want to be supported you can try to send me a couple of RAW files and I can try to create a profile for it. No guarantee it will work because the file format could be changed for new models even for the same manufacturer.

**Comment by ai-gate** [comment]
FunkyRider您好
我目前買的是 Panasonic Lumix DMC-LX10, 但這個傳感器的尺寸是 1 吋感應器的，我記得之在Kent大那個時代，您曾經說過這個部份因傳感器的尺寸太小所以不適合，因沒有太多的細節，
我是有考慮再買一台4/3吋感光元件的尺寸的，但目前尚未決定比較後要買那台比較能夠把RIT的強大細節表示出來。

我指，我有看上Panasonic Lumix LX100 II，但這台已經近四年沒有更新，我也注意到各大廠牌都因為手機拍照的方便性及像素愈來愈高的情況下而導至很少人能拿起相機拍照。

但，我還是喜歡拍照，在您的幫助下，我的Canon 1Dx及1D2都能夠支援RIT的轉換(只有出遠門想帶)，也讓我的拍照一直能很滿意的使用著。

所以我想說，目前因不知道Panasonic Lumix DMC-LX10的RAW檔是否能支援，我怕不能支援的話，就只好再買一台4/3吋感光元件的尺寸的，但怕未來如果又換機的話，一直麻煩您也不好意思，不知道是否能有個請求就是您能否教我怎麼加入新的相機的方式？

我指，我以為只是要在cam_pana及lens_m43加入相機名就好了，但看來不是那麼簡單，不知道您是否有空能教我怎麼加入？付您錢也沒有關係，我不會拿來賣的，我只是單純想一直更新下去新的相機型號而已，謝謝。

這台Panasonic Lumix DMC-LX10 RAW檔可以在這裏下載(麻煩點進去照片裏面在每一個右邊處有RAW可以下載)
https://www.dpreview.com/sample-galleries/2996371353/panasonic-lumix-dmc-lx10-samples-photos

目前看中的Panasonic Lumix LX100 II的RAW檔，但太久沒有更新，所以一直沒有敢買
Panasonic Lumix LX100 II RAW檔可以在這裏下載(麻煩點進去照片裏面在每一個右邊處有RAW可以下載)
https://www.dpreview.com/sample-galleries/7065627980/panasonic-lx100-ii-sample-gallery/

我會想再買台輕便型的相機，但又有支援高光RIT系統的，還請您能夠幫忙，謝謝您。

**Comment by FunkyRider** [comment]
Yes as we have discussed previously, 1" sensor has too much noise even at base ISO (100?) so the highlight detail is barely worth extracting. However it's been several years and things might have improved. I will download the sample and try to come up with camera data that can be used by RITDarkroom and share it here. However I can only do it in the weekends. I will keep you updated.

**Comment by ai-gate** [comment]
好的，那就麻煩您了，謝謝您。

**Comment by ai-gate** [comment]
FunkyRider您好,
不好意思打擾，請問進度好加入嗎？謝謝。

**Comment by FunkyRider** [comment]
Add this section to the end of cam_pana.txt (add a new line below the last line), this will add support for LX10. I have tested with sample RAWs from the link you gave and they look fine to me. LX100-ii doesn't have measurement data on DXO so I can't add it's color transformation matrix.
It's basically a copy-paste of the measurement data from here: https://www.dxomark.com/Cameras/Panasonic/Lumix-DMC-LX10---Measurements where you copy CIE-D50 (day light) and CIE-A (incandescent) "White balance scales" and "Color matrix as defined in ISO standard 17321". You should be able to match the numbers and learn where each one goes and you can add any Panasonic camera on DXO that has measurement data (as long as the RAW format hasn't been changed)
The cam data basically follows this format:

#Camera Name
#0 1 0 4095 (normally this shouldn't change)
#CIE-D50 R G B CIE-A R G B while balance scales
#CIE-D50 Color matrix in column major (Rraw->Rsrgb Graw->Rsrgb Braw->Rsrgb Rraw->Gsrgb Graw->Gsrgb Braw->Gsrgb Rraw->Bsrgb Graw->Bsrgb Braw->Bsrgb)
#CIE-A Color matrix in the same sequence

Hope this helps.

Cam data for LX10:

DMC-LX10
0 1 0 4095
2.44 1 2.13 1.56 1 3.23
1.78 -0.21 0.1 -0.72 1.52 -0.77 -0.06 -0.3 1.67
1.89 -0.28 0.12 -0.82 1.52 -1.13 -0.07 -0.24 2.02

**Comment by FunkyRider** [comment]
I have open sourced the project. The code is available for download and modification for your own use. I used Visual Studio 2012 to build the last v0.49b. If you have newer version, you might have to use its upgrade guide to upgrade the project and fix build errors.

**Comment by ai-gate** [comment]
感謝您，我下載了，我來研究一下，再一次感謝，如果不清楚再上來請教，再一次謝謝您！
